File: python-solutions/problem005.py
from time import time
from functools import reduce
from math import gcd


if __name__ == "__main__":
    start = time()

    answer = int(reduce(
        lambda a, b: a * b / gcd(int(a), 
        int(b)), range(1, 21)
    ))

    end = time() - start

    print(f"\nAnswer for Problem 4: {answer}")
    print( f"Time Taken: {end} seconds\n")

File: golang-solutions/problem005.go
package main

import (
	"fmt"
	"time"
)

func main() {
	start := time.Now()

	answer := 1
	for i := 2; i < 21; i++ {
		answer = LCM(answer, i)
	}

	fmt.Printf("\nAnswer to Problem 5: %d\n", answer)
	fmt.Printf("Time Taken: %v\n\n", time.Since(start))
}

func LCM(a, b int) int {
	return a * b / GCD(a, b)
}

func GCD(a, b int) int {
	for b != 0 {
		t := b
		b = a % b
		a = t
	}
	return a
}
